src/ICSharpCode.SharpZipLib/GZip/GzipOutputStream.cs

```diff
@@ -196,7 +218,8 @@ void WriteHeader()
 					(byte) Deflater.DEFLATED,
 
 					// The flags (not set)
-					0,
+					//0,
+					(byte) (!string.IsNullOrEmpty(_fileName) ?  GZipConstants.FNAME : 0), //AFONSO: File Name
```

Comment:
The flags should be put in an enum that can be cast into a byte here, also remove the commented 0